msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/crosscloud/TestCase1420494.java

```diff
@@ -0,0 +1,133 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+// Acquire token for cross cloud guest account (with broker)
+// https://identitydivision.visualstudio.com/DefaultCollection/IDDP/_workitems/edit/1420494
+@RunWith(Parameterized.class)
+public class TestCase1420494 extends AbstractGuestAccountMsalBrokerUiTest {
+
+    private final String mGuestHomeAzureEnvironment;
+
+    public TestCase1420494(final String name, final String guestHomeAzureEnvironment) {
+        mGuestHomeAzureEnvironment = guestHomeAzureEnvironment;
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Collection guestHomeAzureEnvironment() {
+        return Arrays.asList(new Object[][]{
+                {"AZURE_US_GOV", LabConstants.GuestHomeAzureEnvironment.AZURE_US_GOV},
+                {"AZURE_CHINA_CLOUD", LabConstants.GuestHomeAzureEnvironment.AZURE_CHINA_CLOUD},
+        });
+    }
+
+    /**
+     * Tests Acquiring token for Cross cloud Guest account with broker.
+     */
+    @Test
+    public void test_1420494() throws Throwable {
+        final String userName = mGuestUser.getHomeUpn();
+        final String password = LabGuestAccountHelper.getPasswordForGuestUser(mGuestUser);
+
+        // Handler for Interactive auth call
+        OnInteractionRequired interactionHandler = () -> {
+            PromptHandlerParameters promptHandlerParameters = null;
+            promptHandlerParameters = PromptHandlerParameters.builder()
+                    .prompt(PromptParameter.SELECT_ACCOUNT)
+                    .loginHint(userName)
+                    .staySignedInPageExpected(true)
+                    .broker(mBroker)
+                    .build();
+            AadPromptHandler promptHandler = new AadPromptHandler(promptHandlerParameters);
+            promptHandler.handlePrompt(userName, password);
+        };
+
+        MsalAuthTestParams acquireTokenAuthParams = MsalAuthTestParams.builder()
+                .activity(mActivity)
+                .loginHint(userName)
+                .scopes(Arrays.asList(getScopes()))
+                .promptParameter(Prompt.SELECT_ACCOUNT)
+                .authority(getAuthority())
+                .msalConfigResourceId(getConfigFileResourceId())
+                .build();
+
+        final MsalSdk msalSdk = new MsalSdk();
+        // Acquire token interactively
+        MsalAuthResult acquireTokenResult = msalSdk.acquireTokenInteractive(acquireTokenAuthParams, interactionHandler, TokenRequestTimeout.SHORT);
+        Assert.assertFalse("Verify accessToken is not empty", TextUtils.isEmpty(acquireTokenResult.getAccessToken()));
+
+        // change the time on the device
+        TestContext.getTestContext().getTestDevice().getSettings().forwardDeviceTimeForOneDay();
+
+        // Acquire token silently
+        MsalAuthResult acquireTokenSilentResult = msalSdk.acquireTokenSilent(acquireTokenAuthParams, TokenRequestTimeout.SHORT);
```

Comment:
Since we have removed the throttling at this time..a 30 second wait is not necessary..however, if we put back the throttling (in-memory caching) change back in, then we would need a 30 second wait here to truly test AT expiry because otherwise we would just be getting tokens from throttling cache....to be safe for the future, you may just want to put the 30 second sleep already there just in case we add back throttling in the future (which I do suspect we would do at some point)

Reply:
replied above